src/transfers.py

```diff
@@ -231,8 +231,8 @@ def receive_data(self, s):
             self.current_mtime_usec = s.time.mtime_usec
             if not s.relative_path.startswith(tuple(self.op.top_dir_basenames)):
                 raise ReceiveError("File path is not descended from a valid toplevel directory: %s" % s.relative_path)
-            if self.op.remaining_count == 0:
-                raise ReceiveError("File count exceeds original request size")
+        if self.op.remaining_count == 0:
+            raise ReceiveError("File count exceeds original request size")
```

Comment:
This can probably stay unchanged - if `path != self.current_path`, then a new file is implied, but if the count is already 0 then there shouldn't be any new files. In practice it probably doesn't matter either way - if receive_data is being called when the remaining count is 0, it's invalid no matter what.